novawallet/Modules/AssetsSearch/AssetOperation/AssetOperationNetworkList/AssetOperationNetworkListViewFactory.swift

```diff
@@ -0,0 +1,274 @@
+
+        let wireframe = BuyAssetOperationWireframe(stateObservable: stateObservable)
+
+        return BuyOperationNetworkListPresenter(
+            interactor: interactor,
+            wireframe: wireframe,
+            multichainToken: multichainToken,
+            viewModelFactory: viewModelFactory,
+            selectedAccount: selectedAccount,
+            purchaseProvider: purchaseProvider
+        )
+    }
+}
+
+// MARK: RECEIVE
+
+extension AssetOperationNetworkListViewFactory {
+    static func createReceiveView(
+        with multichainToken: MultichainToken,
+        stateObservable: AssetListModelObservable,
+        selectedAccount: MetaAccountModel
+    ) -> AssetOperationNetworkListViewProtocol? {
+        guard let currencyManager = CurrencyManager.shared else {
+            return nil
+        }
+
+        let operation = OperationQueue()
+        operation.maxConcurrentOperationCount = 1
+
+        let logger = Logger.shared
+
+        let interactor = AssetOperationNetworkListInteractor(
+            multichainToken: multichainToken,
+            operationQueue: operation,
+            stateObservable: stateObservable,
+            logger: logger
+        )
+
+        let presenter = createReceivePresenter(
+            with: interactor,
+            multichainToken: multichainToken,
+            stateObservable: stateObservable,
+            selectedAccount: selectedAccount,
+            currencyManager: currencyManager
+        )
+
+        let view = AssetOperationNetworkListViewController(presenter: presenter)
+
+        presenter.view = view
+        interactor.presenter = presenter
+
+        return view
+    }
+
+    private static func createReceivePresenter(
+        with interactor: AssetOperationNetworkListInteractor,
+        multichainToken: MultichainToken,
+        stateObservable: AssetListModelObservable,
+        selectedAccount: MetaAccountModel,
+        currencyManager: CurrencyManager
+    ) -> ReceiveOperationNetworkListPresenter {
+        let viewModelFactory = createViewModelFactory(with: currencyManager)
+
+        let wireframe = ReceiveAssetOperationWireframe(stateObservable: stateObservable)
+
+        return ReceiveOperationNetworkListPresenter(
+            interactor: interactor,
+            wireframe: wireframe,
+            multichainToken: multichainToken,
+            viewModelFactory: viewModelFactory,
+            selectedAccount: selectedAccount
+        )
+    }
+}
+
+// MARK: SWAPS
+
+extension AssetOperationNetworkListViewFactory {
+    static func createSwapsView(
+        with multichainToken: MultichainToken,
+        stateObservable: AssetListModelObservable,
+        selectClosure: @escaping (ChainAsset) -> Void,
+        selectClosureStrategy: SubmoduleNavigationStrategy
+    ) -> AssetOperationNetworkListViewProtocol? {
+        guard let currencyManager = CurrencyManager.shared else {
+            return nil
+        }
+
+        let operation = OperationQueue()
```

Comment:
It is probably an operationQueue :). And why not to use shared one OperationManagerFacade.sharedQueue?